includes/Modules/Idea_Hub.php

```diff
@@ -225,7 +240,7 @@ private function admin_notice_idea_hub_ideas( $notices ) {
 			return $notices;
 		}
 
-		$transients      = new Transients( $this->context );
+		$transients      = $this->transients;
```

Comment:
No need to assign it to a variable. Just use `$this->transients->...` directly instead of `$transients->...` methods.

Reply:
@eugene-manuilov I assigned it to a variable because the anonymous function has a `use` which I believe can't take `$this->...`? 

```php
'active_callback' => function() use ( $transients, $dismissed_items ) {
	$saved_ideas = $transients->get( self::TRANSIENT_SAVED_IDEAS );
	if ( false === $saved_ideas ) {
		$saved_ideas = $this->get_data( 'saved-ideas' );
		$transients->set( self::TRANSIENT_SAVED_IDEAS, $saved_ideas, DAY_IN_SECONDS );
// ...
```
If I understand it correctly, in a PHP lamdba `$this` is bound to the scope where it is declared rather than when it is called (I guess like lexical `this` in JS)? 
Therefore I wasn't sure when `use` is necessary and when it isn't... in the end I think I just looked at existing examples and copied those.

This also happens further on in the file in `setup_assets` as you mention below. 

So can I remove `$transients` from the `use` in both cases and just reference `$this->transients` inside the lambda?